crates/ethereum/evm/src/execute.rs

```diff
@@ -381,6 +384,39 @@ where
         // NOTE: we need to merge keep the reverts for the bundle retention
         self.state.merge_transitions(BundleRetention::Reverts);
 
+        let mut hashed_state = HashedPostState::from_bundle_state(&self.state.bundle_state.state);
+        let mut state_preimages: HashMap<FixedBytes<32>, Bytes> = HashMap::new();
+
+        for (address, account) in &self.state.cache.accounts {
+            let hashed_address = keccak256(address);
+            hashed_state
+                .accounts
+                .insert(hashed_address, account.account.as_ref().map(|a| a.info.clone().into()));
+
+            let storage = hashed_state
+                .storages
+                .entry(hashed_address)
+                .or_insert_with(|| HashedStorage::new(account.status.was_destroyed()));
+
+            if let Some(account) = &account.account {
+                state_preimages.insert(hashed_address, alloy_rlp::encode(address).into());
+
+                for (slot, value) in &account.storage {
+                    let slot = B256::from(*slot);
+                    let hashed_slot = keccak256(slot);
+                    storage.storage.insert(hashed_slot, *value);
+
+                    state_preimages.insert(hashed_slot, alloy_rlp::encode(slot).into());
+                }
+            }
+        }
+        println!("state_preimages: {:?}", state_preimages);
```

Comment:
I don't think we need to do this inside the executor, this should still occur in `debug_executionWitness` rpc

Reply:
This feels like a natural place to do it?

The plan is to add a `include_preimages` bool in the input and only optionally generate it if configured

Reply:
Or else, we would need to ask block_executor to return `State<DB>` back for debug_execution_witness to be able to access other vars in `State<DB>` like `cache`,

Reply:
Ah, I missed that the current `BlockExecutionOutput` does not have all the information required to include preimages. IMO it would be nice to have separation of concerns, between block execution and fetching preimages though.

One idea is to change the `BlockExecutionOutput` to include the `CacheState`, that way anyone who has a `BlockExecutionOutput` can generate preimages, wdyt?

Reply:
yeah, that works!

Reply:
@Rjected does this look good to you? https://github.com/bluealloy/revm/pull/1756, curious how do we handle revm release cycle here?

Asking because we can directly take the cache out without the convenience method defined in revm, it is a nice to have to keep things consistent.